pkg/agent/apiserver/handlers/bgproute/handler_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bgproute
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+
+	"antrea.io/antrea/pkg/agent/apis"
+	"antrea.io/antrea/pkg/agent/controller/bgp"
+	queriertest "antrea.io/antrea/pkg/querier/testing"
+)
+
+func TestBGPRouteQuery(t *testing.T) {
+	ctx := context.Background()
+	tests := []struct {
+		name             string
+		url              string
+		expectedCalls    func(mockBGPServer *queriertest.MockAgentBGPPolicyInfoQuerier)
+		expectedStatus   int
+		expectedResponse []apis.BGPRouteResponse
+	}{
+		{
+			name: "bgpPolicyState does not exist",
+			expectedCalls: func(mockBGPServer *queriertest.MockAgentBGPPolicyInfoQuerier) {
+				mockBGPServer.EXPECT().GetBGPRoutes(context.Background(), true, true).Return(nil, bgp.ErrBGPPolicyNotFound)
+			},
+			expectedStatus: http.StatusNotFound,
+		},
+		{
+			name: "get all advertised routes",
+			expectedCalls: func(mockBGPServer *queriertest.MockAgentBGPPolicyInfoQuerier) {
+				mockBGPServer.EXPECT().GetBGPRoutes(ctx, true, true).Return(
+					[]string{"192.168.1.0/24", "192.168.2.0/24", "fec0::10:96:10:10/128"}, nil)
+			},
+			expectedStatus: http.StatusOK,
+			expectedResponse: []apis.BGPRouteResponse{
+				{
+					Route: "192.168.1.0/24",
+				},
+				{
+					Route: "192.168.2.0/24",
+				},
+				{
+					Route: "fec0::10:96:10:10/128",
+				},
+			},
+		},
+		{
+			name: "get advertised ipv4 routes only",
+			url:  "?ipv4-only=",
```

Comment:
I would expect `antctl` to generate `?ipv4-only` (no equal sign). Is this not the case? If it is true, it's better to omit the equal sign in test cases.

Reply:
i checked the value in `r.URL.RawQuery`, it is with equal sign. So, used the same in testcases